tests/integration/proxy/testproxymain.cpp

```diff
@@ -511,6 +511,28 @@ void TestProxyImpl::on_test_signal_boolean_cb(const bool s) {
     record_signal();
 }
 
+void TestProxyImpl::on_notification_received()
+{
+    m_pending_notifications--;
+    if (m_pending_notifications == 0) {
+        check_done();
+    }
+}
+
+void TestProxyImpl::expect_notification(sigc::signal<void> &signal)
+{
+    m_pending_notifications++;
+    signal.connect(sigc::mem_fun(this, &TestProxyImpl::on_notification_received));
+}
+
+void TestProxyImpl::check_done()
```

Comment:
How does this go together with the asynchronous method calls? Do we want e.g. a `m_pending_methods` counter as well?

I guess the test works now because things are processed in order and method calls are performed before signal and property tests. But... yeah... I don't know. Something you have thought about?

Reply:
Hum, thinking about it some more... A counter for methods is probably good to have as well, otherwise you can not be sure that we have churned through the whole method chain. Will still emit done (and exit with success) if just the notifications and signals work.

Reply:
I added one more commit for this.

Reply:
Nice! :)